lang_qc/db/helper/well.py

```diff
@@ -350,13 +350,24 @@ def assign_qc_state(
             db_state.is_preliminary = values["is_preliminary"]
             db_state.user = values["user"]
             db_state.created_by = values["created_by"]
-            db_state.date_updated = values["date_updated"]
-            values["date_created"] = db_state.date_created  # Save for historic.
+            if date_updated:
+                db_state.date_updated = date_updated
         else:
-            values["date_created"] = values["date_updated"]
-            db_state = QcState(**values)  # Create a new record.
+            db_state = QcState(
+                date_created=date_updated, date_updated=date_updated, **values
+            )  # Create a new record.
```

Comment:
Would be nice to comment that using undefined `date_updated` does work, ie triggers setting the default value by the db